source/dub/recipe/packagerecipe.d

```diff
@@ -171,50 +213,139 @@ struct ConfigurationInfo {
 	}
 }
 
+/// Actual representation of a dependency as specified by the user
+private struct DependencyObject
+{
+	@Name("version")
+	@Optional string version_;
+	@Optional string path;
+	@Optional string repository;
+	bool optional;
+	@Name("default")
+	bool default_;
+
+	@Name("dflags", true) string[][string] dflags;
```

Comment:
it's not obvious what this bool parameter is. Also doesn't seem to be in upstream configy, should this maybe be a separate UDA? (e.g. `@StartsWith("...")`)

Could also just make that return a Name struct with the second field set, but it'd be better on call sites.

Reply:
Yeah I had to add support for the suffix approach in Configy to support dub's scheme.
One of the requirement built into the library was that field names are static. There are two built-in exceptions:
- The `Key` UDA:
https://github.com/dlang/dub/blob/044cf5e94681261ac86a237ae8da93ddf5cecba0/source/configy/Attributes.d#L50-L88
- Fully dynamic names can be achieved with `string[string]` (although I'm still on the fence whether or not that was a good idea);

The weird scheme that dub uses, with dashes, needed special support because I couldn't just allow any names, otherwise it would mean disabling strict mode for the whole `dub.json`. The best compromise was to add this boolean option in the UDA that only matches the beginning of the name. I don't think adding an extra UDA is worth it because I don't think we should upstream this.

Reply:
> I don't think adding an extra UDA is worth it because I don't think we should upstream this.

Whatever is in dub repo, must match upstream. That is one of my biggest concerns from the get-go for this line of work. Let's not introduce a fork.

A UDA would of course be better than a random sentinel boolean. Plus someone else may need this capability.

Reply:
> Whatever is in dub repo, must match upstream.

Why ? We don't have this requirement anywhere. I'm not saying we shouldn't have this requirement, but that is not something we currently enforce / have discussed. And I think it should be its own discussion.

The package recipe format is currently too tied to JSON. The prefix approach is one of those limitation, and I really want to change the format, down the line, and remove this configy special-case.

Reply:
I'm just asking to add a helper function for the Name UDA here, so:

```d
Name StartsWith(string name) { return Name(name, true); }
```

so in the code we can use `@StartsWith("name")`

Reply:
I'm saying about upstream due to vibecompat. It is just adding more maintenance to a unique copy.

Breaking people's dub files needlessly is going to have pushback including from myself.

Reply:
@WebFreak001 : Ah that makes sense, will do.

@rikkimax : We're not breaking *any* recipe files, quite the opposite. When I mentioned changing the format, it's a long term plan that won't be enacted any time soon, and if/when it is, it will be both backward compatible and include a whole lot of change (and can be automated).